modules/cudev/include/opencv2/cudev/functional/functional.hpp

```diff
@@ -668,6 +668,33 @@ template <typename T, typename D> struct saturate_cast_func : unary_function<T,
     }
 };
 
+// Convert Fp16 dummy
+template <typename T, typename D> struct saturate_cast_fp16_func : unary_function<T, D>
```

Comment:
I think it will be better to leave the class unimplemented for the default case. In this case, compiler will raise an error, if someone tries to use it with unsupported types:

```
template <typename T, typename D> struct saturate_cast_fp16_func;

template <> struct saturate_cast_fp16_func<float, short> : unary_function<float, short>
{
...
};
```